### PR TITLE
feat: calculates Staker's pending rewards

### PR DESCRIPTION
**Pull request by impelcrypto**
**Pull Request Summary**

* feat: calculates Staker's pending rewards

**Check list**

- [ ] contains breaking changes
- [x] adds new feature
- [ ] modifies existing feature (bug fix or improvements)
- [ ] relies on other tasks
- [ ] documentation changes
- [ ] tested on mobile devices

**Release notes:**

* feat: calculates Staker's pending rewards

**This pull request makes the following changes:**

**Adds**

* feat: calculates Staker's pending rewards
image

**To-dos**
* Move `pending-rewards.ts` to Astar.js

**Note**
The amount of each era's token issuance is used to calculate the total amount of unclaimed rewards. In order to minimize fetching of this value from on-chain, the function assumes the amount of 1 era's issuance by taking the average of 7 eras. Hence the amount of unclaimed rewards is slightly different from the actual amount. 
In other words, as the number of unclaimed eras increases, the difference increases (but I don't think it's too far off).

### CODE REVIEW

**Comment by Kahonnohak** [comment]
Amazing work!　People will love this:)

There was a 2.3% margin of error when I tried. So
- Lets say the title is **Estimated Rewards**
- Tooltip (it has not been changed since v1...); `The rewards displayed are an estimate, and there may be a small difference between what is shown and what you actually receive.`
- Please place (era) on the right

**Comment by impelcrypto** [comment]
> There was a 2.3% margin of error when I tried

This is because the UI doesn't display amounts with decimals. According to my calculations, the difference was 0.04%.

Estimated rewards: 177.725
Actual rewards: 177.645

177.725 - 177.645 = 0.08
0.08 / 177.645 = 0.00045034
0.00045034 * 100 = 0.04%

image

Screenshot 2023-04-03 at 1 49 44 PM

**Comment by impelcrypto** [comment]
@Kahonnohak Updated.

**Comment by impelcrypto** [comment]
Hi @bobo-k2 @sirius651 @gluneau @niklabh Please help review this PR. I'd like to include this change into the next version update of the portal.

**Comment by impelcrypto** [comment]
@niklabh @bobo-k2 @sirius651 @gluneau  
> To-dos
    Move pending-rewards.ts to Astar.js

I've moved `pending-rewards.ts` to Astar.js. Please help take a look at this [PR](https://github.com/AstarNetwork/astar.js/pull/46) as well. 🙏

**Review by sirius651 — APPROVED** [review]
